src/jer/de.rs

```diff
@@ -0,0 +1,583 @@
+        &mut self,
+        _t: crate::Tag,
+        _c: Constraints,
+    ) -> Result<Utf8String, Self::Error> {
+        decode_jer_value!(Self::string_from_value, self.stack)
+    }
+
+    fn decode_visible_string(
+        &mut self,
+        _t: crate::Tag,
+        _c: Constraints,
+    ) -> Result<VisibleString, Self::Error> {
+        decode_jer_value!(Self::string_from_value, self.stack)?
+            .try_into()
+            .map_err(|e| {
+                DecodeError::string_conversion_failed(
+                    Tag::VISIBLE_STRING,
+                    alloc::format!("Error transforming VisibleString: {e:?}"),
+                    crate::Codec::Jer,
+                )
+            })
+    }
+
+    fn decode_general_string(
+        &mut self,
+        _t: crate::Tag,
+        _c: Constraints,
+    ) -> Result<GeneralString, Self::Error> {
+        decode_jer_value!(Self::string_from_value, self.stack)?
+            .try_into()
+            .map_err(|e| {
+                DecodeError::string_conversion_failed(
+                    Tag::GENERAL_STRING,
+                    alloc::format!("Error transforming GeneralString: {e:?}"),
+                    crate::Codec::Jer,
+                )
+            })
+    }
+
+    fn decode_ia5_string(
+        &mut self,
+        _t: crate::Tag,
+        _c: Constraints,
+    ) -> Result<Ia5String, Self::Error> {
+        decode_jer_value!(Self::string_from_value, self.stack)?
+            .try_into()
+            .map_err(|e| {
+                DecodeError::string_conversion_failed(
+                    Tag::IA5_STRING,
+                    alloc::format!("Error transforming IA5String: {e:?}"),
+                    crate::Codec::Jer,
+                )
+            })
+    }
+
+    fn decode_printable_string(
+        &mut self,
+        _t: crate::Tag,
+        _c: Constraints,
+    ) -> Result<PrintableString, Self::Error> {
+        decode_jer_value!(Self::string_from_value, self.stack)?
+            .try_into()
+            .map_err(|e| {
+                DecodeError::string_conversion_failed(
+                    Tag::PRINTABLE_STRING,
+                    alloc::format!("Error transforming PrintableString: {e:?}"),
+                    crate::Codec::Jer,
+                )
+            })
+    }
+
+    fn decode_numeric_string(
+        &mut self,
+        _t: crate::Tag,
+        _c: Constraints,
+    ) -> Result<NumericString, Self::Error> {
+        decode_jer_value!(Self::string_from_value, self.stack)?
+            .try_into()
+            .map_err(|e| {
+                DecodeError::string_conversion_failed(
+                    Tag::NUMERIC_STRING,
+                    alloc::format!("Error transforming NumericString: {e:?}"),
+                    crate::Codec::Jer,
+                )
+            })
+    }
+
+    fn decode_teletex_string(
+        &mut self,
+        _t: crate::Tag,
+        _c: Constraints,
+    ) -> Result<TeletexString, Self::Error> {
+        todo!()
```

Comment:
Why todo here?

Reply:
Currently we do not have any logic to convert between `String` and `TeletexString` or to validate a `TeletexString`'s bytes and it doesn't seem to be a trivial addition to make. PER also has a `todo!()` in its `Decoder` impl, so I'd propose to tackle this issue in a separate PR.

Reply:
Just a note when out of curiosity looking this PR, Teletex standard has been withdrawn. Should we just delete the type? https://www.itu.int/rec/T-REC-T.61 https://en.wikipedia.org/wiki/ITU_T.61 and focus on other more used ones. Get back to it if someone really needs it.

Reply:
Unfortunately [it keeps sticking around](https://www.itu.int/ITU-T/formal-language/itu-t/x/x509/2019/PkiPmiExternalDataTypes.html) 🥲  It is also used in `rasn-pkix`. However, I think we can afford to be lenient when it comes to the implementations in `jer`, `xer`, `oer`, or `per`, since AFAIK T.61 had already been revoked when they were conceived.

Reply:
Oh, I missed that one. It is mentioned in `oer` as (TeletexString, T61String) both, but I haven't faced any standard which actually uses the types yet.

Reply:
This might be good reference if we end up implementing it:

https://github.com/mouse07410/asn1c/blob/84d3a59c1bb89c59be6ca0625bb14ebea9084ba5/skeletons/TeletexString.c#L4